### PR TITLE
added DraftJS editor to HtmlField with Draft's content to HTML converter

### PR DESCRIPTION
**Pull request by garfieldduck**
added DraftJS editor to HtmlField in order to replace TinyMCE editor in the near future

### CODE REVIEW

File: fields/types/html/DraftHtmlConverter.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	});
+
+	// end tag with </ol> or </ul>: or if it is the last block
+	if (raw.blocks.length > 0 && nestedTagMap[raw.blocks[raw.blocks.length - 1].type]) {
+		html += nestedTagMap[nestLevel.shift()][1];         // close with </ol> or </ul>
+	}
+
+	return html;
+
+}
```
Comment (nickhsine) [inline fields/types/html/DraftHtmlConverter.js]: would you like make DraftHtmlConverter as an object, not a function.
make ```DraftHtmlConverter.convert``` be the function.
or just rename this file as ```convertDraftToHtml```.

File: fields/types/html/HtmlField.js [hunk 2]
```diff
@@ -15,53 +16,40 @@ function getId () {
...
-				this.initWysiwyg();
+		catch (error) {
+			// transform existing HTML content produced by TinyMCE
+			if (this.props.value && typeof (this.props.value) === 'string') {
+				// create an EditorState from HTML
+				editorState = EditorState.createWithContent(ContentState.createFromBlockArray(processHTML(this.props.value)));
 			} else {
-				this.initWysiwyg();
+				// create empty EditorState
+				editorState = EditorState.createEmpty();
```
Comment (nickhsine) [inline fields/types/html/HtmlField.js]: If we use Draft from the scratch, maybe we don't need to care about those posts made by tinyMCE.

File: fields/types/html/HtmlField.js [hunk 3]
```diff
@@ -70,116 +58,97 @@ module.exports = Field.create({
...
-		} else {
-			return;
-		}
-
-		this._currentValue = content;
+	onChange (editorState) {
 		this.props.onChange({
 			path: this.props.path,
-			value: content,
+			value: this.state.value,
```
Comment (nickhsine) [inline fields/types/html/HtmlField.js]: should ```value``` be defined in ```getInitialState``` ?

File: fields/types/html/HtmlField.js [hunk 4]
```diff
@@ -70,116 +58,97 @@ module.exports = Field.create({
...
-			Object.assign(options.additionalOptions, importcssOptions);
-		}
+		const content = Draft.convertToRaw(editorState.getCurrentContent());
+		const cHtml = DraftHtmlConverter(content);
+		this.setState({ editorState });
 
-		if (!options.overrideToolbar) {
-			toolbar += ' | code';
+		// set value if the content has been changed
+		if (this.state.html !== cHtml) {
```
Comment (nickhsine) [inline fields/types/html/HtmlField.js]: should ```this.state.html``` be ```this.state.valuue.html``` ?

File: fields/types/html/HtmlField.js [hunk 5]
```diff
@@ -70,116 +58,97 @@ module.exports = Field.create({
...
+			this.setState({ value: {
+				draft: content,
+				html: cHtml,
+			} });
+			this.setState({ draftStr: JSON.stringify(this.state.value) });
```
Comment (nickhsine) [inline fields/types/html/HtmlField.js]: should ```draftStr```be in the ```getInitialState``` ?

File: fields/types/html/HtmlType.js [hunk 6]
```diff
@@ -20,5 +20,118 @@ util.inherits(html, FieldType);
...
+	var schema = this.list.schema;
+
+	var paths = this.paths = {
+		// md: this._path.append('.md'),
+		draft: this._path.append('.draft'),
+		// draftHistory: this._path.append('.draftHistory'),
+		html: this._path.append('.html'),
+	};
+
+	var markedOptions = this.markedOptions;
```
Comment (nickhsine) [inline fields/types/html/HtmlType.js]: could this line be removed?

File: fields/types/html/HtmlType.js [hunk 7]
```diff
@@ -20,5 +20,118 @@ util.inherits(html, FieldType);
...
+	// var setDraft = function (val) {
+	// 	if (val && val.draft && val.html) {
+	// 		console.log("\n setDraft", val.draft, typeof(val.draft));
+	// 		console.log("\n setHtml", val.html);
+	// 		this.set(paths.draft, JSON.stringify(val.draft));
+	// 		this.set(paths.html, val.html);
+	// 	}
+	// 	return JSON.stringify(val.draft);
+	// };
+	console.log("this.path", this.path);
```
Comment (nickhsine) [inline fields/types/html/HtmlType.js]: remove comments and console

File: fields/types/markdown/MarkdownType.js [hunk 8]
```diff
@@ -133,6 +134,7 @@ markdown.prototype.isModified = function (item) {
 
 markdown.prototype.updateItem = function (item, data, callback) {
 	var value = this.getValueFromData(data);
+
 	if (value !== undefined) {
 		item.set(this.paths.md, value);
 	} else if (this.paths.md in data) {
```
Comment (nickhsine) [inline fields/types/markdown/MarkdownType.js]: revert this change ?

File: fields/types/html/HtmlField.js (hunk 2 above)
Comment (garfieldduck) [inline fields/types/html/HtmlField.js]: I will remove all the related statements

**Comment by nickhsine** [comment]
nice work :+1: :100:
